### PR TITLE
NPC system ShopModule now shows itens which can be bought and selled.

### PR DESCRIPTION
**Pull request by joseivan**
NPC shop module does not give player a way to know what NPC sells or buys so added this capability. Player just need to say buy or sell with no parameters.

### CODE REVIEW

File: data/npc/lib/npcsystem/modules.lua [hunk 1]
```diff
@@ -273,7 +273,7 @@ if Modules == nil then
 	end
 
 	function KeywordModule:parseKeywords(data)
-		local n = 1
+		local n = 1
```
Comment (WibbenZ) [inline data/npc/lib/npcsystem/modules.lua]: Delete the tab

File: data/npc/lib/npcsystem/modules.lua [hunk 2]
```diff
@@ -561,10 +563,37 @@ if Modules == nil then
 		if ret ~= nil then
 			self:parseBuyableContainers(ret)
 		end
+
+		if #self.buyableItens > 0 then
+
```
Comment (WibbenZ) [inline data/npc/lib/npcsystem/modules.lua]: Delete

File: data/npc/lib/npcsystem/modules.lua [hunk 3]
```diff
@@ -561,10 +563,37 @@ if Modules == nil then
...
+								"buy"
+							}, 
+							ShopModule.showBuyableItens,
+							{	
+								module = self
+							})
+		end
+
+		if #self.sellableItens > 0 then
+
```
Comment (WibbenZ) [inline data/npc/lib/npcsystem/modules.lua]: Delete

File: data/npc/lib/npcsystem/modules.lua [hunk 4]
```diff
@@ -561,10 +563,37 @@ if Modules == nil then
...
+							{
+								"sell"
+							}, 
+							ShopModule.showSellableItens,
+							{	
+								module = self
+							})
+		end
+
+
```
Comment (WibbenZ) [inline data/npc/lib/npcsystem/modules.lua]: Delete aswell as the one above

File: data/npc/lib/npcsystem/modules.lua [hunk 5]
```diff
@@ -561,10 +563,37 @@ if Modules == nil then
...
 	end
 
 	-- Parse a string contaning a set of buyable items.
 	function ShopModule:parseBuyable(data)
+
```
Comment (WibbenZ) [inline data/npc/lib/npcsystem/modules.lua]: Delete

File: data/npc/lib/npcsystem/modules.lua [hunk 6]
```diff
@@ -608,6 +637,8 @@ if Modules == nil then
 				end
 			else
 				if name ~= nil and itemid ~= nil and cost ~= nil then
+
```
Comment (WibbenZ) [inline data/npc/lib/npcsystem/modules.lua]: Delete

**Comment by WibbenZ** [comment]
Delete all the empty lines, I would try to write the arrays abit cleaner aswell.

**Comment by marksamman** [comment]
Thanks for the pull request, but we're not interested in this functionality in the default data pack.

**Pull request closed without merging**